src/drivers/nuttx/lv_nuttx_touchscreen.c

```diff
@@ -85,38 +87,70 @@ lv_indev_t * lv_nuttx_touchscreen_create(const char * dev_path)
  *   STATIC FUNCTIONS
  **********************/
 
-static void touchscreen_read(lv_indev_t * drv, lv_indev_data_t * data)
+static void conv_touch_sample(lv_indev_t * drv,
+                              lv_indev_data_t * data,
+                              struct touch_sample_s * sample)
 {
     lv_nuttx_touchscreen_t * touchscreen = drv->driver_data;
-    struct touch_sample_s sample;
+    uint8_t touch_flags = sample->point[0].flags;
 
-    /* Read one sample */
+    if(touch_flags & (TOUCH_DOWN | TOUCH_MOVE)) {
+        lv_display_t * disp = lv_indev_get_display(drv);
+        int32_t hor_max = lv_display_get_horizontal_resolution(disp) - 1;
+        int32_t ver_max = lv_display_get_vertical_resolution(disp) - 1;
 
-    int nbytes = read(touchscreen->fd, &sample,
-                      sizeof(struct touch_sample_s));
+        data->point.x = LV_CLAMP(0, sample->point[0].x, hor_max);
+        data->point.y = LV_CLAMP(0, sample->point[0].y, ver_max);
+        touchscreen->last_state = LV_INDEV_STATE_PRESSED;
+    }
+    else if(touch_flags & TOUCH_UP) {
+        touchscreen->last_state = LV_INDEV_STATE_RELEASED;
+    }
+}
 
-    /* Handle unexpected return values */
+static bool touchscreen_read_sample(int fd, struct touch_sample_s * sample)
+{
+    int nbytes = read(fd, sample, sizeof(struct touch_sample_s));
+    return nbytes == sizeof(struct touch_sample_s);
+}
 
-    if(nbytes == sizeof(struct touch_sample_s)) {
-        uint8_t touch_flags = sample.point[0].flags;
+static void touchscreen_read(lv_indev_t * drv, lv_indev_data_t * data)
+{
+    lv_nuttx_touchscreen_t * touchscreen = drv->driver_data;
+    struct touch_sample_s sample;
 
-        if(touch_flags & TOUCH_DOWN || touch_flags & TOUCH_MOVE) {
-            const lv_display_t * disp_drv = drv->disp;
-            int32_t ver_max = disp_drv->ver_res - 1;
-            int32_t hor_max = disp_drv->hor_res - 1;
+    /* Note: Since it is necessary to avoid multi-processing click events
+    * caused by redundant continue_reading, a two-unit sample sliding window
+    * algorithm is used here. continue_reading is only activated when there
+    * are two points in the window.
+    */
 
```

Comment:
```suggestion
    /* 
     * Note: Since it is necessary to avoid multi-processing click events
     * caused by redundant continue_reading, a two-unit sample sliding window
     * algorithm is used here. continue_reading is only activated when there
     * are two points in the window.
     */

```